src/WCSimConstructPMT.cc

```diff
@@ -146,15 +146,14 @@ G4LogicalVolume* WCSimDetectorConstruction::ConstructPMT(G4String PMTName, G4Str
 
   //logicGlassFaceWCPMT->SetVisAttributes(G4VisAttributes::Invisible);
   logicGlassFaceWCPMT->SetVisAttributes(WCPMTVisAtt);
-
-  if (!aWCPMT)
-  {
-    G4SDManager* SDman = G4SDManager::GetSDMpointer();
-    G4String SDName = "/WCSim/";
-    SDName += CollectionName;
-    aWCPMT = new WCSimWCSD(CollectionName,SDName,this );
-    SDman->AddNewDetector( aWCPMT );
-  }
+ 
+  // Instantiate a new sensitive detector and register this sensitive detector volume with the SD Manager. 
+  G4SDManager* SDman = G4SDManager::GetSDMpointer();
+  G4String SDName = "/WCSim/";
+  SDName += CollectionName;
+  aWCPMT = new WCSimWCSD(CollectionName,SDName,this );
```

Comment:
It looks like aWCPMT is now being overwritten each time a new PMT is created. Will this cause a problem? Probably not if aWCPMT is not used anywhere else. If that's the case, we should move it from a member variable to a local variable in this function. If it turns out we do need to use aWCPMT, we probably need to make it a map keyed to the CollectionName.